### PR TITLE
237 add with cart validation

### PR DESCRIPTION
**Pull request by sepowitz**
This PR introduces the withCartValidation logic and applies it to the setOrderLocationId action for the time being. Was planning on adding it to ```setRequestedAt``` and ```setServiceType``` in a separate PR, but the internal logic and overall pattern should be all here.

I tested it pretty heavily against the Tartine codebase and it's working quite well!

I wasn't sure how else to handle the ```unmetDeliveryMinimum``` case, as there isn't much we can do from brandibble-redux apart from surface the error from Brandibble and let the client handle it.

### CODE REVIEW

File: src/actions/session/order.js [hunk 1]
```diff
@@ -360,56 +361,90 @@ export function validateCurrentCart(brandibble, data = {}) {
...
+  return dispatch => {
     const { orders } = brandibble;
     const order = orders.current();
     const payload = orders.validate(order, data).then(res => res);
     return dispatch(_validateCurrentOrder(payload));
   };
 }
 
-export function setOrderLocationId(currentOrder, locationId) {
+export function setOrderLocationId(
```
Comment (sepowitz) [inline src/actions/session/order.js]: See here

File: src/actions/session/order.js [hunk 2]
```diff
@@ -525,3 +560,158 @@ export function submitOrder(brandibble, order, options = {}) {
...
+export function setServiceType(currentOrder, serviceType, onValidationError) {
+  return dispatch => {
+    return dispatchEvent(_withCartValidation({ location_id: locationId }, () => innerLogic))
+  }
+}
+
+ *
+ */
+
+export function _withCartValidation(
```
Comment (sepowitz) [inline src/actions/session/order.js]: See here

File: src/utils/constants.js [hunk 3]
```diff
@@ -5,6 +5,14 @@ export const Status = {
   REJECTED: 'REJECTED',
 };
 
+export const ErrorCodes = {
```
Comment (sepowitz) [inline src/utils/constants.js]: See here

File: src/utils/getInvalidLineItems.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import get from './get';
+
+export default (invalidItems, lineItemsData) =>
```
Comment (sepowitz) [inline src/utils/getInvalidLineItems.js]: See here

File: src/actions/session/order.js [hunk 5]
```diff
@@ -525,3 +560,134 @@ export function submitOrder(brandibble, order, options = {}) {
...
+                    setRequestedAt(orderRef, firstAvailableOrderTime),
+                  ).then(dispatch(actionCallback()));
+                });
+              }
+
+              /**
+               * Unmet delivery minimum
+               * (not much we can do here apart from notify the customer)
+               */
+              if (errorCode === ErrorCodes.validateCart.unmetDeliveryMinimum) {
```
Comment (sepowitz) [inline src/actions/session/order.js]: Not sure what else we can do here, apart from return the brandibble error.

File: src/utils/constants.js (hunk 3 above)
Comment (hhff) [reply src/utils/constants.js]: Are these available to the host application? ie exported from brandibble-redux itself

**Comment by sepowitz** [comment]
> fantastic work. can u just ensure that the host app can import `ErrorCodes` and then let's merge it

Yup, def!

**Pull request merged**